core_impl/src/main/java/io/opencensus/implcore/stats/MeasureToViewMap.java

```diff
@@ -84,10 +86,13 @@ private synchronized MutableViewData getMutableViewData(View.Name viewName) {
   }
 
   /** Enable stats collection for the given {@link View}. */
-  synchronized void registerView(View view, Clock clock) {
+  synchronized void registerView(View view, List<? extends Handler> handlers, Clock clock) {
```

Comment:
Would it be better to process the list of handlers in another class, and have MeasureToViewMap just hold the map of aggregated data?  It seems like the code that manages the handlers could use just the export intervals, handlers, and registered views to determine what views to query at different times.  Then it could call getView on the StatsManager or MeasureToViewMap and pass the results to the handlers.

Reply:
My idea is, if a view is registered using a push model (i.e with a list of handlers), we need to block the thread where the Stats library is running, until we make sure view is successfully registered against the given handlers.That's because if we don't block the thread, ViewData might be exported to handlers before view is registered for that handler. This is hard to achieve if we move the logic on registering view for handlers to other class.

Reply:
I don't think that that race condition is possible if the new export class controls exporting to handlers.  `ViewManagerImpl.registerView(View view, List<? extends Handler> handlers)` could add the view to MeasureToViewMap and then call a method on the new export class to give it the view and handlers.  Then the new export class could determine when to export data to each handler and get that data from MeasureToViewMap when necessary.